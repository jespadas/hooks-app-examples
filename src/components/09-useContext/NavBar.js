import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (

        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">

            <Link to="/" className="navbar-brand">useContext</Link>

            <div className="navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">


                    <NavLink
                        exact
                        activeClassName="active"
                        to="/"
                        className="nav-link"
                    >
                        Home
                    </NavLink>



                    <NavLink
                        exact
                        activeClassName="active"
                        to="/about"
                        className="nav-link"
                    >
                        About
                    </NavLink>

                    <NavLink
                        exact
                        activeClassName="active"
                        to="/login"
                        className="nav-link"
                    >
                        Login
                    </NavLink>


                </ul>
            </div>
        </nav>

    )
}
