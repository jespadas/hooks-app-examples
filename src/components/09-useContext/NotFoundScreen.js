import React from 'react';
import { Link } from 'react-router-dom';

export const NotFoundScreen = () => {
    return (
        <div>
            <h1> 404 - Not Found !</h1>
            <Link to="/">
                Go back home !
            </Link>
        </div>
    )
}
