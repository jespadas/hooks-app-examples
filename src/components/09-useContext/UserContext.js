import { createContext } from 'react';

/*
    useContext hook is used to create common data that can be accessed 
    throughout the component hierarchy without passing the props down 
    manually to each level. Context defined will be available to all 
    the child components without involving “props”.
*/

export const UserContext = createContext(null);
