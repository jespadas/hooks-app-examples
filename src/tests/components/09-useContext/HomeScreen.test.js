import React from 'react';
import { mount } from 'enzyme';

import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Test on <HomeScreen />', () => {

    const user = {
        name: "Julio",
        email: "contact@julioespadas.com"
    };

    /*  Mount is a method that re - mounts the component, 
        if it is not currently mounted.This can be
        used to simulate a component going through
        an unmount / mount lifecycle. 
    */

    const wrapper = mount(
        <UserContext.Provider value={{
            user: user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('Should be displayed correctly', () => {

        expect(wrapper).toMatchSnapshot();

    })


})
