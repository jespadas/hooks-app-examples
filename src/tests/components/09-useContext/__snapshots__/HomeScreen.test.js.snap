// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <HomeScreen /> Should be displayed correctly 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
   "name": "Julio",
   "email": "contact@julioespadas.com"
}
  </pre>
</div>
`;
